lib/ecmascript.ts

```diff
@@ -2637,7 +2705,23 @@ export function TimeZoneEquals(one: string | Temporal.TimeZoneProtocol, two: str
   if (one === two) return true;
   const tz1 = ToTemporalTimeZoneIdentifier(one);
   const tz2 = ToTemporalTimeZoneIdentifier(two);
-  return tz1 === tz2;
+  if (tz1 === tz2) return true;
+  const offsetMinutes1 = ParseTimeZoneIdentifier(tz1).offsetMinutes;
+  const offsetMinutes2 = ParseTimeZoneIdentifier(tz2).offsetMinutes;
+  if (offsetMinutes1 === undefined && offsetMinutes2 === undefined) {
+    // Calling GetAvailableNamedTimeZoneIdentifier is costly, so (unlike the
+    // spec) the polyfill will early-return if one of them isn't recognized. Try
```

Comment:
Seems reasonable, and would be an editorial change AFAICT. @ptomato what do you think?

Note that in native implementations this operation is vastly less expensive. For native implementations, `GetAvailableNamedTimeZoneIdentifier` is: 
1. Convert input identifier to ASCII lower case
2. Find that input in a list or hashtable of all ~600 named time zone identifiers
3. If found, return the primary time zone identifier string corresponding to that input string
4. Otherwise, return NULL.

Reply:
That probably wouldn't be something I'd bother putting in the spec, since native implementations are likely to do this quite differently as you point out, but putting this in the reference polyfill seems fine.

Reply:
OK, no changes then. I also remembered something: it should be quite rare for an input time zone to not be found here. It should only happen for custom time zones or garbage input data. So even if implementations do bother to optimize, it won't make a difference in normal cases. So I think we can resolve this one.